src/hotspot/os/windows/os_windows.cpp

```diff
@@ -867,21 +868,60 @@ int os::active_processor_count() {
     return ActiveProcessorCount;
   }
 
+  bool schedules_all_processor_groups = false;
+  if (UseAllWindowsProcessorGroups) {
+    // Starting with Windows 11 and Windows Server 2022, the OS has changed to
+    // make processes and their threads span all processors in the system,
+    // across all processor groups, by default. Therefore, we will allow all
+    // processors to be active processors on these operating systems. However,
+    // job objects can be used to restrict processor affinity across the
+    // processor groups. In this case, the number of active processors must be
+    // obtained from the processor affinity in the job object.
+    schedules_all_processor_groups = win32::is_windows_11_or_greater() || win32::is_windows_server_2022_or_greater();
+    if (schedules_all_processor_groups) {
+      DWORD processors_in_job_object = win32::active_processors_in_job_object();
```

Comment:
Shouldn't the check for a job object be unconditional? I don't understand what the connection is to either `UseAllWindowsProcessorGroups` or the Windows version here. ??

Reply:
> Shouldn't the check for a job object be unconditional? I don't understand what the connection is to either `UseAllWindowsProcessorGroups` or the Windows version here. ??

A job object can enable multiple Windows processor groups. I think running with -XX:-UseAllWindowsProcessorGroups should override the job object enabling processors in >1 processor group (closer to behavior that would be seen in older OpenJDK versions). The approach I've taken adds a warning when this happens. Other than that, I've updated the implementation to respect the number of processors enabled by the job object. On older Windows versions, this includes displaying a warning that only 1 processor group is supported. Please let me know what you think of this approach.